src/Sentry/SentryClient.cs

```diff
@@ -121,22 +121,69 @@ public void CaptureTransaction(Transaction transaction)
                                 "to properly finalize the transaction and send it to Sentry.");
         }
 
-
         // Sampling decision MUST have been made at this point
-        Debug.Assert(transaction.IsSampled != null,
+        Debug.Assert(transaction.IsSampled is not null,
             "Attempt to capture transaction without sampling decision.");
 
-
-        if (transaction.IsSampled != true)
+        if (transaction.IsSampled is false)
         {
             _options.ClientReportRecorder.RecordDiscardedEvent(DiscardReason.SampleRate, DataCategory.Transaction);
             _options.LogDebug("Transaction dropped by sampling.");
             return;
         }
 
+        var processedTransaction = BeforeSendTransaction(transaction);
+        if (processedTransaction is null) // Rejected transaction
+        {
+            // TODO: should the dropped transaction be recorded?
```

Comment:
@mattjohnsonpint, should dropped transactions be recorded?

Reply:
Yes.  Use `DiscardReason.BeforeSend` and `DataCategory.Transaction`.